forward/forwarder_test.go

```diff
@@ -65,17 +61,43 @@ type Pong struct {
 }
 
 func (s *ForwarderTestSuite) registerPong(address string, channel *tchannel.Channel) {
-	handler := func(ctx json.Context, ping *Ping) (*Pong, error) {
-		return &Pong{"Hello, world!", address}, nil
+	hmap := map[string]interface{}{
+		"/ping": func(ctx json.Context, ping *Ping) (*Pong, error) {
+			return &Pong{"Hello, world!", address}, nil
+		},
+		"/error": func(ctx json.Context, ping *Ping) (*Pong, error) {
+			return nil, errors.New("remote error")
+		},
 	}
-	hmap := map[string]interface{}{"/ping": handler}
+	s.Require().NoError(json.Register(channel, hmap, func(ctx context.Context, err error) {}))
+
+	thriftHandler := &pingpong.MockTChanPingPong{}
 
-	s.Require().NoError(json.Register(channel, hmap,
-		func(ctx context.Context, err error) {}))
+	// successful request
+	thriftHandler.On("Ping", mock.Anything, &pingpong.Ping{
+		Key: "success",
+	}).Return(&pingpong.Pong{
+		Source: address,
+	}, nil)
+
+	// error request
+	thriftHandler.On("Ping", mock.Anything, &pingpong.Ping{
+		Key: "error",
+	}).Return(nil, &pingpong.PingError{})
+
+	server := thrift.NewServer(channel)
+	server.Register(pingpong.NewTChanPingPongServer(thriftHandler))
 }
 
 func (s *ForwarderTestSuite) SetupSuite() {
-	s.sender = &DummySender{"127.0.0.1:3001", "127.0.0.1:3001"}
+	sender := &MockSender{}
```

Comment:
This name and the implementation is code generated by mockery based on the name of the interface that is used. I don't know if Sender is the best name for an interface here. I will have a look to see if I can come up with a better name. The `Mock` part will at least stay in the name here because of mockery.